From 155a29259c9d7e6939e531b8ba1cbbe03bb17576 Mon Sep 17 00:00:00 2001
From: Xu Mao <B34688@freescale.com>
Date: Wed, 5 Mar 2014 13:04:55 +0800
Subject: [PATCH 62/72] ENGR00133115 cts FlakyNetwork test

add bNetworkFail flag in OMXPlayer. Don't send PLAYBACK_COMPLETE
to java when EOS occurs with bNetworkFail is true

correct current status according to player status

Signed-off-by: Xu Mao <B34688@freescale.com>
---
 include/media/OMXPlayer.h      | 1 +
 media/libmedia/mediaplayer.cpp | 5 +++++
 2 files changed, 6 insertions(+)

diff --git a/include/media/OMXPlayer.h b/include/media/OMXPlayer.h
index 293f9cd..dceff34 100644
--- a/include/media/OMXPlayer.h
+++ b/include/media/OMXPlayer.h
@@ -112,6 +112,7 @@ private:
     status_t            getTrackInfo(Parcel *reply);
 
     bool                qdFlag;
+    bool                bNetworkFail;
 };
 
 class OMXPlayerType
diff --git a/media/libmedia/mediaplayer.cpp b/media/libmedia/mediaplayer.cpp
index 0f6d897..00f12d5 100644
--- a/media/libmedia/mediaplayer.cpp
+++ b/media/libmedia/mediaplayer.cpp
@@ -352,6 +352,11 @@ bool MediaPlayer::isPlaying()
             ALOGE("internal/external state mismatch corrected");
             mCurrentState = MEDIA_PLAYER_PAUSED;
         }
+        if ((mCurrentState & MEDIA_PLAYER_PLAYBACK_COMPLETE) && temp) {
+            ALOGE("internal/external state mismatch corrected");
+            mCurrentState = MEDIA_PLAYER_STARTED;
+        }
+        
         return temp;
     }
     ALOGV("isPlaying: no active player");
-- 
1.8.0

